tests/test_controller.py

```diff
@@ -25,3 +25,66 @@ def test_init_stac(mocap_nwb, stac_config, rodent_config):
     assert stac.model_cfg == model_cfg
     assert stac._kp_names == sorted_kp_names
     assert isinstance(stac._mj_model, _structs.MjModel)
+
+
+def test_align_joint_dims():
+    from jax import numpy as jp
+
+    types = [0, 1, 2, 3]
+    ranges = [[0.0, 0.0], [0.0, 1.0], [-1.0, 1.0], [-1.0, 1.0]]
```

Comment:
Ball joint has four coordinates, but we only list a single min, max pair. Is there a reason for that? Is there a way to specify a ball joint limited to an asymmetric cone?

Reply:
I don't think so. Documentation on ball joint range:

>  For ball joints, the limit is imposed on the angle of rotation (relative to the reference configuration) regardless of the axis of rotation. Only the second range parameter is used for ball joints; the first range parameter should be set to 0.